payment-request/PaymentItem/type_member.https.html

```diff
@@ -0,0 +1,76 @@
+<!doctype html>
+<meta charset="utf8">
+<link rel="help" href="https://w3c.github.io/payment-request/#dom-paymentitem-type">
+<title>
+  PaymentItem type member
+</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+const validMethods = [
+  { supportedMethods: "basic-card" },
+  { supportedMethods: "https://apple.com/apple-pay" },
+];
+const validTotal = {
+  label: "Total",
+  amount: {
+    currency: "USD",
+    value: "5.00",
+  },
+};
+const validDisplayItem = {
+  label: "Item",
+  amount: {
+    currency: "USD",
+    value: "1.00",
+  },
+};
+const validDetails = {
+  total: validTotal,
+  displayItems: [validDisplayItem],
+};
+
+test(() => {
+  new PaymentRequest(validMethods, validDetails);
+}, "Smoke test");
+
+test(() => {
+  // Let's make an invalid DisplayItem for the total
+  const invalidTotal = { ...validTotal, type: "this is not valid" };
+  const invalidDetails = {
+    ...validDetails,
+    total: invalidTotal,
+  };
+  assert_throws(new TypeError(), () => {
+    new PaymentRequest(validMethods, invalidDetails);
+  });
+}, "An invalid enum value for PaymentDetailsInit.total's type throws TypeError");
+
+test(() => {
+  // Let's make an invalid DisplayItem to add to displayItems
+  const invalidDisplayItem = { ...validDisplayItem, type: "this is not valid" };
```

Comment:
Since Edge doesn't support object spreading I guess this should change along with the other places?

Reply:
yeah :( There are a few others that will need the same treatment. ITS IE6 ALL OVER AGAIN! 🤣